### PR TITLE
Implement ledger node cache.

### PR DESCRIPTION
**Pull request by mattcollier**

### CODE REVIEW

**Comment by mattcollier** [comment]
@dlongley @gannan08 This is ready for review/merging.  Ganesh's testing has show it to be beneficial in ledger deployments.

File: lib/rootApi.js [hunk 1]
```diff
@@ -314,6 +305,8 @@ api.remove = callbackify(async (actor, ledgerNodeId, options = {}) => {
       translate: 'owner'
     });
 
+  // delete cache entry even if the update operation fails, the cache will be
```
Comment (gannan08) [inline lib/rootApi.js:308]: I am failing to find where the cache entry is being deleted? Is this comment relevant?

File: lib/rootApi.js [hunk 2]
```diff
@@ -14,6 +14,9 @@ const {validate} = require('bedrock-validation');
 const validator = require('./validator');
 const LedgerNode = require('./LedgerNode');
 
+const {default: {LruMemoize}} = require('@digitalbazaar/lru-memoize');
+const lruMemoize = new LruMemoize({max: 1000});
```
Comment (gannan08) [inline lib/rootApi.js:18]: Is there a default `maxAge`? It seems that we are asserting that it's safe to cache the ledgerNode document indefinitely. If that is the case, can we capture that in a comment that this is intended behavior.

File: lib/rootApi.js (hunk 1 above)
Comment (mattcollier) [reply lib/rootApi.js:311]: good catch, fixed.

File: lib/rootApi.js (hunk 2 above)
Comment (mattcollier) [reply lib/rootApi.js]: another good catch, on second thought, it's not safe, added a 5 min `maxAge`.

**Review by gannan08 — APPROVED** [review]

File: lib/rootApi.js [hunk 3]
```diff
@@ -387,3 +385,34 @@ api.getNodeIterator = callbackify(async (actor, options = {}) => {
...
+      'Ledger node not found.',
+      'NotFound',
+      {httpStatusCode: 404, ledger: ledgerNodeId, public: true});
+  }
+  // only need to check permissions if the ledger is owned, otherwise it
+  // is public
+
+  // FIXME: is it safe to make this if(actor && record.ledgerNode.owner)
+  // what are important distinctions between undefined and null in this case?
+  if(record.ledgerNode.owner) {
```
Comment (dlongley) [inline lib/rootApi.js:408]: This should be safe:

```suggestion
  if(actor !== null && record.ledgerNode.owner) {
```

If an `actor` is undefined, the actor is anonymous, which is different from if `actor` is `null` -- in which case the permission check is skipped. Of course, this is a terrible distinction :), but it's legacy code... and once everything is moved to a capabilities based architecture we won't have to care about it anymore.

**Review by dlongley — APPROVED** [review]
See my comment about the FIXME. Thanks!

**Pull request merged**